openbas-api/src/test/java/io/openbas/service/ReportServiceTest.java

```diff
@@ -0,0 +1,131 @@
+package io.openbas.service;
+
+import io.openbas.database.model.Report;
+import io.openbas.database.model.ReportInformation;
+import io.openbas.database.model.ReportInformationsType;
+import io.openbas.database.repository.ReportRepository;
+import io.openbas.rest.report.form.ReportInformationInput;
+import io.openbas.rest.report.form.ReportInput;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+public class ReportServiceTest {
+
+    @Mock
+    private ReportRepository reportRepository;
+
+    private ReportService reportService;
+
+    @BeforeEach
+    void before() {
+        // Injecting mocks into the controller
+        reportService = new ReportService(reportRepository);
```

Comment:
Mocking the repository is one of the approaches in unit testing, as it isolates the service layer and allows you to test business logic without involving the actual database, making tests faster and more focused. Additionally, since the repository is a Spring Boot layer, it has already been tested.

Reply:
OK, I see
Thanks for the explanations, ok for me